libraries/AP_BattMonitor/AP_BattMonitor.h

```diff
@@ -128,13 +141,19 @@ class AP_BattMonitor
     bool overpower_detected(uint8_t instance) const;
 
     // cell voltages
-    const cells & get_cell_voltages() { return get_cell_voltages(AP_BATT_PRIMARY_INSTANCE); };
+    bool has_cell_voltages() { return has_cell_voltages(AP_BATT_PRIMARY_INSTANCE); }
+    bool has_cell_voltages(const uint8_t instance) const;
+    const cells & get_cell_voltages() const { return get_cell_voltages(AP_BATT_PRIMARY_INSTANCE); }
     const cells & get_cell_voltages(const uint8_t instance) const;
 
     // temperature
     bool get_temperature(float &temperature) const { return get_temperature(temperature, AP_BATT_PRIMARY_INSTANCE); };
     bool get_temperature(float &temperature, const uint8_t instance) const;
 
+    // get battery resistance estimate in ohms
+    float get_resistance() const { return get_resistance(AP_BATT_PRIMARY_INSTANCE); }
+    float get_resistance(uint8_t instance) const { return state[instance].resistance; }
```

Comment:
Nitpicking: to be more correct and match other methods make _instance_  method parameter _const_